src/shared/apollo.js

```diff
@@ -6,16 +6,21 @@ import fetch from 'isomorphic-unfetch';
 
 let cachedClient;
 
-const createApolloClient = () =>
-  new ApolloClient({
+const createApolloClient = () => {
+  const { NEXT_PUBLIC_GITHUB_ACCESS_TOKEN } = process.env;
```

Comment:
Having this exposed client side is a security issue as well. I know NextJS has some server API capabilities that I haven't had the chance to play with yet. I think ideally I'd proxy all requests through the NextJS server to Github so the keys are never publicly exposed. In a different ecosystem, I don't think this would be as big a problem since auth would be handled differently but definitely something I'm not thrilled about here.